src/index.ts

```diff
@@ -203,6 +203,26 @@ function resolveLaravelPlugin(pluginConfig: Required<PluginConfig>): LaravelPlug
             process.on('SIGHUP', process.exit)
 
             exitHandlersBound = true
+
+            return () => {
+                server.middlewares.use((req, res, next) => {
+                    if (req.url === '/index.html') {
+                        server.config.logger.warn(
+                            "\n" + colors.bgYellow(
+                                colors.black(`The Vite server should not be accessed directly. Please visit ${appUrl} instead.`)
```

Comment:
Should this be less directive and informative instead to match the HTML page?

```suggestion
                                colors.black(`The Vite server should not be accessed directly. Your Laravel application's configured APP_URL is: ${appUrl}`)
```